test/ArchToken.js

```diff
@@ -0,0 +1,67 @@
+// We import Chai to use its asserting functions here.
+const { expect } = require("chai");
+const exp = require("constants");
+
+describe("Arch Token test suit", function () {
+    // parseUnits uses 18 decimal by default
+    const tokenSupply = ethers.utils.parseUnits("100");
+    console.log(tokenSupply);
+
+    let contract;
+    let token;
+    let owner;
+    let user1;
+    let user2;
+    let users;
+
+    beforeEach(async function () {
+        contract = await ethers.getContractFactory("ArchToken");
+        [owner, user1, user2, ...users] = await ethers.getSigners();
+        token = await contract.deploy();
+        await token.mint(owner.address, tokenSupply);
+    });
+
+    describe("Minting", function () {
+        it("Should be able to mint to specific address", async function () {
+            expect(await token.balanceOf(owner.address)).to.eq(tokenSupply);
+        });
+    });
+
+    // basic end-to-end testing of underlying erc20
+    describe("Transactions", function () {
+        it("Should not be able to transfer more than total supply", async function () {
+            let amount = ethers.utils.parseUnits("101");
+            await expect(
+                token.transfer(user1.address, amount)
+            ).to.be.revertedWith("ERC20: transfer amount exceeds balance");
+        });
+
+        it("Should be able to transfer tokens between accounts", async function () {
+            let amount = ethers.utils.parseUnits("1");
```

Comment:
With the pre-mint change - this one becomes a bit more tricky. 

I would also recommend testing calling transfer from accounts that are not an owner or the Treasury (look for `connect`) that means we'll also test that `approve` works

Reply:
Will add "transferFrom" functionality and associated tests to check this